src/CycleGraph.jl

```diff
@@ -0,0 +1,170 @@
+
+# =======================================================
+#         CycleGraph struct
+# =======================================================
+
+"""
+    CycleGraph <: AbstractGraph
+
+A structure representing an undirected cycle graph.
+
+A `CycleGraph` with one vertex is a single vertex without any edges (no self-loops)
+and a `CycleGraph` with two vertices is a single edge.
+
+See also: [`LightGraphs.cycle_graph`](@ref)
+"""
+struct CycleGraph{T<:Integer} <: LG.AbstractGraph{T}
+    nv::T
+
+    function CycleGraph{T}(nv) where {T}
+       
+        nv = convert(T, nv)
+        nv >= zero(T) || throw(ArgumentError("nv must be >= 0"))
+
+        return new{T}(nv)
+    end
+end
+
+CycleGraph(nv) = CycleGraph{typeof(nv)}(nv)
```

Comment:
```suggestion
CycleGraph(nv::T) where {T<: Integer} = CycleGraph{T}(nv)
```

it might be me, but I prefer using `typeof` in signatures only when necessary (functions mostly)